[FIX] l10n_hu_edi: missing invoice number in new layouts

RicGR98 opened the pull request:
Since the introduction of new layouts (e.g. folder), the
invoice number is not displayed in Hungary. This is because
the xpath is incorrect because the new layouts now have
two children instead of one.

The old ones only had one child node:

https://github.com/odoo/odoo/blob/4dad76c045f94861632929189baa96d69f662275/addons/web/views/report_templates.xml#L478-L506

Whereas the new ones have two:

https://github.com/odoo/odoo/blob/4dad76c045f94861632929189baa96d69f662275/addons/web/views/report_templates.xml#L535-L550

https://github.com/odoo/odoo/blob/4dad76c045f94861632929189baa96d69f662275/addons/web/views/report_templates.xml#L551-L579

So the problem with the current xpath is that it's replacing
both children, even though only the second one should be.
By 'replacing' the first one (by iffing it),
`layout_document_title` is not displayed anymore and therefore
no invoice number appears on the invoice.

To fix this, we use a different xpath on the `img` parent.
For consistency and maintainability reasons, we apply this
new xpath on all the layouts and not just the new ones.

opw-4397438

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/192650.png)](https://mergebot.odoo.com/odoo/odoo/pull/192650)

RicGR98 commented on addons/l10n_hu_edi/views/report_invoice.xml:
```diff
@@ -114,14 +113,4 @@
         </xpath>
     </template>
 
-    <template id="custom_header">
```
Wooops right, sorry ! It's pushed again :)

jco-odoo commented on addons/l10n_hu_edi/views/report_templates.xml:
```diff
@@ -2,6 +2,12 @@
     <template id="external_layout_standard" inherit_id="web.external_layout_standard" >
+        <!-- hide company_address as it is taken care of in report_invoice.xml -->
+        <xpath expr="//div[@name='company_address']" position="replace">
+            <t t-if="o and o._name == 'account.move' and o.company_id.account_fiscal_country_id.code == 'HU'"/>
```
Could not that be done with a "custom_header2" (or no_header_hu e.g.) then?  I am just afraid that we have other reports on account.move.

RicGR98 commented:
@jco-odoo I've simplified the code to modify the xpath's instead of creating new ones, which makes the diff easier and cleaner than adding/removing a new template. I've updated the commit message accordingly. What do you think ?

jco-odoo commented:
@robodoo r+

robodoo commented:
@RicGR98 @jco-odoo 'ci/runbot' failed on this reviewed PR.
